### PR TITLE
FAQ: update checklist from perlbug and reference github issue tracker

### PR DESCRIPTION
**Pull request by Grinnz**
The checklist is from a very outdated version of perlbug's docs, and now we need to direct people to the Github tracker rather than perlbug.

### CODE REVIEW

File: src/misc/cpan-faq.html [hunk 1]
```diff
@@ -1038,147 +1042,178 @@ <h3>
...
+        <p>
+            Note that reports about bugs in old versions of Perl, especially
+            those which indicate you haven't also tested the current stable
+            release of Perl, are likely to receive less attention from the
+            volunteers who build and maintain Perl than reports about bugs in
+            the current release.
+        </p>
+        <p>
+            This tool isn't appropriate for reporting bugs in any version prior
+            to Perl 5.0.
```
Comment (rspier) [inline src/misc/cpan-faq.html:1064]: Drop this paragraph, the odds of someone reporting a bug in perl4 now are pretty slim.

File: src/misc/cpan-faq.html [hunk 2]
```diff
@@ -1038,147 +1042,178 @@ <h3>
...
         </p>
     </li>
     <li>
         <b>Can you fix the bug yourself?</b>
         <p>
-            A bug report which includes a patch to fix it will almost
-            definitely be fixed. Use the diff program to generate your patches
-            (diff is being maintained by the GNU folks as part of the diffutils
-            package, so you should be able to get it from any of the GNU
-            software repositories). If you do submit a patch, the cool-dude
```
Comment (rspier) [inline src/misc/cpan-faq.html:1134]: Sad to see this paragraph leave :)

Comment (rspier) [reply src/misc/cpan-faq.html:1134]: (Don't interpret this as a reason to keep it.)

File: src/misc/cpan-faq.html [hunk 3]
```diff
@@ -1029,7 +1033,7 @@ <h3>
 </p>
 <p>
     Here is a checklist from <a href=
-    "http://perldoc.perl.org/perlbug.html">perlbug</a>, a bug reporting tool
+    "https://perldoc.perl.org/perlbug">perlbug</a>, a bug reporting tool
```
Comment (rspier) [inline src/misc/cpan-faq.html:1036]: Long term, would it be better to just remove this section from here and link directly to the list?  (i.e. reduce duplication)

**Review by rspier — APPROVED** [review]
Very minor comments below.

File: src/misc/cpan-faq.html (hunk 1 above)
Comment (Grinnz) [reply src/misc/cpan-faq.html:1064]: Probably a good idea to drop it from the source docs as well.

File: src/misc/cpan-faq.html (hunk 3 above)
Comment (Grinnz) [reply src/misc/cpan-faq.html:1036]: Possibly, though the best we could do is link to https://perldoc.perl.org/perlbug#DESCRIPTION - and frankly these docs very rarely change and are even less likely to do so now

**Pull request merged**